pkg/controller/clusters/controlplane.go

```diff
@@ -101,6 +101,25 @@ func ReconcileControlPlane(ctx context.Context, c client.Client, cp *ControlPlan
 		clientutil.AddAnnotation(cp.KubeadmControlPlane, clusterv1.PausedAnnotation, "true")
 	}
 
+	// When the controller reconciles the control plane for a cluster with an external etcd configuration
+	// the KubeadmControlPlane.Spec.KubeadmConfigSpec.ClusterConfiguration.Etcd.External.Endpoints field is
+	// defaulted to an empty slice. However, at some point that field in KubeadmControlPlane object is filled
+	// and updated by another component
+
+	// We do not want to update the field with an empty slice again, so here we check if the endpoints for the
+	// external etcd have already been populated on the KubeadmControlPlane object and override ours before applying it.
+	kcp := &controlplanev1.KubeadmControlPlane{}
+	kcpKey := client.ObjectKeyFromObject(cp.KubeadmControlPlane)
```

Comment:
we already have the capi cluster
can we use the reference it has?

Reply:
yes, we can